packages/emitter-framework/src/java/components/enum.tsx

```diff
@@ -0,0 +1,76 @@
+import { mapJoin } from "@alloy-js/core";
+import * as jv from "@alloy-js/java";
+import { Enum } from "@typespec/compiler";
+
+export interface EnumProps {
+  type: Enum;
+}
+
+export function EnumDeclaration({ type }: EnumProps) {
+  const members = Array.from(type.members.values());
+
+  let valueType: string | undefined;
+  let isConsistent = true;
+
+  members.forEach((member, index) => {
+    if (member.value !== undefined && member.value !== null) {
+      const currentType = typeof member.value === "string" ? "String" : "int";
+
+      if (valueType === undefined) {
+        valueType = currentType;
+      } else if (valueType !== currentType) {
+        isConsistent = false;
+      }
+    }
+  });
+
+  // TypeSpec allows you to mix and match values, for a Java enum have to have one value
+  if (!isConsistent) {
+    throw new Error("All enum values must have the same type");
```

Comment:
If we decide to throw error, we should have the error message be more specific. Include the enum type name and the value types.